data/lua/core/audio.lua

```diff
@@ -0,0 +1,30 @@
+
+if wesnoth.kernel_type() == "Game Lua Kernel" then
+	-- Only deprecation stubs for now
+	wesnoth.play_sound = wesnoth.deprecate_api('wesnoth.play_sound', 'wesnoth.audio.play', 1, nil, wesnoth.audio.play)
+	wesnoth.sound_volume = wesnoth.deprecate_api('wesnoth.sound_volume', 'wesnoth.audio.volume', 1, nil, function(volume)
+		local old_volume = wesnoth.audio.volume
+		if type(volume) == 'number' then
+			wesnoth.audio.volume = volume
+		end
+		return old_volume
+	end)
+	wesnoth.add_sound_source = wesnoth.deprecate_api('wesnoth.add_sound_source', 'wesnoth.audio.sources', 1, nil, function(cfg)
+		wesnoth.audio.sources[cfg.id] = cfg
+	end, 'Assign a config to the sound source ID')
+	wesnoth.get_sound_source = wesnoth.deprecate_api('wesnoth.get_sound_source', 'wesnoth.audio.sources', 1, nil, function(id)
+		return wesnoth.audio.sources[id]
+	end, 'Index by the sound source ID')
+	wesnoth.remove_sound_source = wesnoth.deprecate_api('wesnoth.remove_sound_source', 'wesnoth.audio.sources', 1, nil, function(id)
+		wesnoth.audio.sources[id] = nil
+	end, 'Assign nil to the sound source ID')
+	wesnoth.music_list = wesnoth.deprecate_api('wesnoth.music_list', 'wesnoth.audio.music_list', 1, nil, setmetatable({}, {
```

Comment:
Given #4459, #4030 and #4458, is wesnoth.audio.music_list ready? It seems there's going to be further changes to that API.

Reply:
I'm not sure if there will be changes, but it's ready for use in its current form.

Reply:
IIUC from the IRC discussion, a reason for accepting a lot of deprecations now is to reduce the number of deprecations between 1.16 and 1.18. This sounds as if bits of the newly-added API will get deprecated before 1.18 is released.

Reply:
Well, as far as I know there are no current plans to make any of the API changes noted in those issues, so I don't think it makes sense to hold off on this change on the grounds that someone _might_ hypothetically make more changes in the future (a future which might not even _be_ in the 1.17 cycle - it might end up waiting for 1.19 for example).

Reply:
@Pentarctagon please would you give a decision about this?

Reply:
What I want to avoid in 1.17 and onward is deprecating something simply to rename it, move it, re-order its arguments, or any other sort of superficial change, since 1.15 is already a lost cause in that regard anyway.  If an API needs to have a breaking change made to it because the current implementation is actually functionally broken somehow, then I think that's fine inasmuch as it's better to have a working API.

Reply:
So renaming it before 1.16 and then replacing it with a different set of functions after 1.16 is accepted? Okay, I've misunderstood then.

Reply:
I'm not sure how that would be different than renaming?

Reply:
The proposal in the bug reports more or less comes down to "we need to replace our use of SDL_Mixer entirely", which would be a lot of work and which nobody has said they have any intention of doing.

The argument "we shouldn't to <thing> because someone else might at some point maybe do something else" isn't a very convincing one to me. While I do think there's too much API churn, accepting that argument would make it very hard to do a lot of things, or anything at all if taken to its extreme.